core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java

```diff
@@ -404,15 +425,17 @@ private int declineContractOffers() {
             if (response.isCompletedExceptionally()) {
                 process.transitionDeclining();
                 negotiationStore.save(process);
+                invokeForEach(l -> l.declining(process));
                 monitor.debug(format("[Consumer] Failed to send contract rejection. ContractNegotiation %s stays in state %s.",
                         process.getId(), ContractNegotiationStates.from(process.getState())));
                 continue;
```

Comment:
Not introduced by this PR but could this code be refactored to avoid the `continue` statement? 
Pretty old-fashioned and dangerous.